### ISSUE TITLE
TxPool: Transactions is validated prior to EIP1559 Normalization

### ISSUE THREAD

**Issue by jangko** [status-im/nimbus-eth1#1021]
in tx_packer.nim

```Nim
proc vmExecGrabItem(pst: TxPackerStateRef; item: TxItemRef)
  ...

  if not xp.classifyValidatePacked(vmState, item):         # <---- the call to validateTransaction is in here
    return ok(false) # continue with next account

  let
    accTx = vmState.stateDB.beginSavepoint 
    gasUsed = pst.runTx(item)                              # < ---- EIP1559 Tx normalization is in here
```

On the other hand, in `process_transaction.nim`, EIP1559 normalization is done *before* calling `validateTransaction`.

log message:
```text
DBG 2022-03-30 11:35:26.817+07:00 processed jobs topics="tx-pool"
  tid: 9936
  file: tx_pool.nim:624
  nJobs: 1

(pending: 0, staged: 1, packed: 0, total: 1, disposed: 0)
DBG 2022-03-30 11:35:26.820+07:00 invalid tx: maxFee is smaller than baseFee
  tid: 9936
  file: validate.nim:282
  maxFee: 0
  baseFee: 875000000
```

Because of this, I think every legacy tx will never reach packed bucket and will stay in staged bucket because they will never pass transaction validation when we are assembling post london block. 

@mjfh  can you please take a look at this? I'm not sure if I move the EIP1559 tx normalization out of `runTx` is safe, maybe something else need to be moved too. Thank you.

**Comment by mjfh** [comment]
> (pending: 0, staged: 1, packed: 0, total: 1, disposed: 0)
> DBG 2022-03-30 11:35:26.820+07:00 invalid tx: maxFee is smaller than baseFee
>   tid: 9936
>   file: validate.nim:282
>   maxFee: 0
>   baseFee: 875000000

Certainly, this has to be fixed. IMHO it is an error. If I see it right it is a problem in `validateTransaction()` checking post London fields for a legacy TX? Alternatively the `baseFee` could be reset in `classifyValidatePacked()` for legacy transactions.

I guess one should adapt both.

**Comment by jangko** [comment]
> If I see it right it is a problem in validateTransaction() checking post London fields for a legacy TX?

it is not a problem if we normalize the tx according to EIP1559 rule. `maxFee` will be set to `gasPrice` which is usually bigger than `baseFee`. my concern is will there any serious problem if EIP1559 tx normalization moved before `validateTransaction()`?

**Comment by jangko** [comment]
fixed by #1019
